R/utils.r

```diff
@@ -124,7 +124,19 @@ first_upper <- function(x) {
 
 download <- function(path, url, ...) {
   request <- httr::GET(url, ...)
-  httr::stop_for_status(request)
+
+  if (httr::status_code(request) >= 400) {
+    # Parse the response if it is from GitHub, use stop_for_status otherwise.
+    tryCatch(github_response(request),
+      error = function(e) {
+        if (inherits(e, "github_response")) {
```

Comment:
If you did go this route (which I don't think we should), it would be better to write:

```R
tryCatch(
   ...,
   github_response = function(e) stop(e),
   error = function(e) ...
)
```